jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ServletCoreResponse.java

```diff
@@ -333,7 +336,10 @@ public Mutable add(HttpHeader header, String value)
         @Override
         public Mutable add(HttpField field)
         {
-            _response.addHeader(field.getName(), field.getValue());
+            if (field.getHeader() == HttpHeader.CONTENT_LENGTH)
+                _response.setContentLengthLong(field.getLongValue());
+            else
+                _response.addHeader(field.getName(), field.getValue());
```

Comment:
```suggestion
            if (field.getHeader() == HttpHeader.CONTENT_LENGTH)
                _response.setContentLengthLong(field.getLongValue());
            _response.addHeader(field.getName(), field.getValue());
```

In the Content-Length specific case, we don't want to drop the normal response.addHeader too.  right?

Reply:
@joakime The setContentLengthLong does the addHeader, but with a long value included, so we don't have to parse the string to work out the content length.

Reply:
Gotcha.
When testing the `org.eclipse.jetty.ee9.servlet.CrossContextDispatcherTest.testForwardToResourceHandler` with just the `abort(x)` change alone it seems to be happy.

The Content-Length and HttpOutput.written length efforts don't seem to be related.
What other test cases are these fixing?

Reply:
The content-length change was just an improvement notice whilst debugging the comparison between the bytes written and the content-length set.   I noticed we were using the addHeader mechanism that was doing integer to string and string to integer conversion when it was not necessary.  So this change is not a fix, just an optimization.